core/src/variables.rs

```diff
@@ -1,28 +1,175 @@
+use crate::CoreError;
 use serde::{Deserialize, Serialize};
 use std::cmp::Ordering;
+use std::collections::hash_map::Entry;
+use std::collections::{HashMap, HashSet};
+use std::fmt;
+use std::hash::Hash;
 
-// This is, for now, a placeholder file for variables. The data types it contains is (some) of the
-// once it will contain when the variable store is actually implemented.
+#[derive(Debug, Clone, Default)]
+pub struct VariableStore(HashMap<String, Value>);
 
-// The type of a variable
+impl VariableStore {
+    pub fn get(&self, name: &str) -> Option<&Value> {
+        self.0.get(name)
+    }
+
+    /// Clears the variable store
+    pub fn clear(&mut self) {
+        self.0.clear()
+    }
+
+    /// Ensure that a valid name is used.
+    /// Only ASCII alphanumerics characters and "_" is allowed. The name may also not start with a number.
+    fn valid_name(&self, name: &str) -> Result<(), CoreError> {
+        // Ensure that the first character is not a number and that the name
+        // is at least 1 character long
+        if name.chars().next().map(char::is_numeric).unwrap_or(true) {
```

Comment:
Here, you are creating the same iterator twice. It seems unneccecary. I think we can do somthing like this:
```rust
let mut chars = name.chars();
let valid = chars.next().map(|c| c.is_ascii_alphabetic() || c == '_').unwrap_or_default() && chars.all(|c| c.is_ascii_alpanumeric() || c == '_');
```
or something like that, to make sure we don't creat the iterator twice. Maybe doesn't have that large of an impact anyhow